helm/chart-operator/templates/deployment.yaml

```diff
@@ -41,11 +41,11 @@ spec:
       serviceAccountName: {{ tpl .Values.resource.default.name  . }}
       {{- if .Values.chartOperator.cni.install }}
       hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
```

Comment:
Not sure I agree with this change. The label might change with the rename to `control-plane` and running on control-planes caused trouble before IIRC.

Reply:
there is no way around this.
This operating mode is meant to be run during a cluster bootstrap, when CNI and kube-proxy might not be running.
The only way for chart operator to talk to the API server is through 127.0.01 when running inside a master node.

Also, the goal should be to make chart-operator switch to default mode as soon as flux takes over the chart-operator App.

Reply:
Actually, no.
In CAPI world it's staying in the `cni.install = true` mode all the time.

See https://github.com/giantswarm/config/blob/main/default/apps/chart-operator/configmap-values.yaml.template#L14 and https://github.com/giantswarm/config/blob/main/installations/gohan/apps/chart-operator/configmap-values.yaml.patch#L3

I think this is very wrong and should be fixed, but that's not this PR's responsibility IMHO